### PR TITLE
merge dev branch

### PR DESCRIPTION
**Pull request by lgarithm**

### CODE REVIEW

**Comment by luomai** [comment]
let me know when this PR is ready to review.

**Comment by luomai** [comment]
By far, it looks good to me.

**Comment by lgarithm** [comment]
It is ready for review.

File: src/misaka.h [hunk 1]
```diff
@@ -17,10 +17,7 @@ struct dtypes_t {
     const uint8_t f64;
 };
 
-// TODO: define dtypes const struct
-// extern dtypes_t dtypes;
-
-#define DTYPE_FLOAT 0x0D
+extern const dtypes_t dtypes;
```
Comment (luomai) [inline src/misaka.h]: what does dtypes_t mean?

Comment (lgarithm) [reply src/misaka.h]: it is used to simulate namespace in C, so users can use `dtypes.f32` instead of macros.

**Pull request merged**